lib/signaling/v2/peerconnection.js

```diff
@@ -1745,6 +1745,14 @@ function filterOutMediaStreamIds(sdp) {
  * @returns {boolean}
  */
 function shouldRecycleTransceiver(transceiver, pcv2) {
+  // NOTE(lrivas): Disable transceiver recycling for Safari to prevent
+  // payload type reuse conflicts during track unpublish/republish scenarios
+  // in P2P rooms. See, https://github.com/twilio/twilio-video.js/issues/2122.
+  // This can be revisited once we remove the SDP munging logic for Safari.
+  if (isSafari) {
+    return false;
```

Comment:
Does recycling the transceiver cause the PT clash? I can see how previously assigned PTs might get reused unaware of the recycling.